site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/citations/CitationsSearch.js

```diff
@@ -0,0 +1,161 @@
+// This file is part of Zenodo.
+// Copyright (C) 2022 CERN.
+//
+// Zenodo is free software; you can redistribute it
+// and/or modify it under the terms of the GNU General Public License as
+// published by the Free Software Foundation; either version 2 of the
+// License, or (at your option) any later version.
+//
+// Zenodo is distributed in the hope that it will be
+// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Zenodo; if not, write to the
+// Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+// MA 02111-1307, USA.
+//
+// In applying this license, CERN does not
+// waive the privileges and immunities granted to it by virtue of its status
+// as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+import React from "react";
+import { PropTypes } from "prop-types";
+import _truncate from "lodash/truncate";
+import {
+  ReactSearchKit,
+  InvenioSearchApi,
+  ResultsLoader,
+  SearchBar,
+  Error,
+  Pagination,
+  ResultsPerPage,
+  EmptyResults,
+} from "react-searchkit";
+import { OverridableContext } from "react-overridable";
+import { Grid, Container } from "semantic-ui-react";
+import _get from "lodash/get";
+import { Results } from "./components/Results";
+import { Filter } from "./components/Filter";
+import { NoCitations } from "./components/NoCitations";
+
+const ZENODO_BROKER_ENDPOINT =
+  "https://zenodo-broker.web.cern.ch/api/relationships";
+
+const resultsPerPageValues = [
+  { text: "10", value: 10 },
+  { text: "20", value: 20 },
+  { text: "30", value: 30 },
+];
+
+const overriddenComponents = {
+  "citationsSearch.BucketAggregation.element": ({ containerCmp }) =>
+    containerCmp,
+  "citationsSearch.BucketAggregationContainer.element": ({ valuesCmp }) =>
+    valuesCmp,
+  "citationsSearch.EmptyResults.element": NoCitations,
+};
+
+export class CitationsSearch extends React.Component {
+  initialState = () => {
+    const { recordPIDs } = this.props;
+    const doi = _get(recordPIDs, "doi.identifier");
+    return {
+      sortBy: "mostrecent",
+      sortOrder: "asc",
+      layout: "list",
+      page: 1,
+      size: 10,
+      filters: [
+        ["group_by", "version"],
+        ["id", doi],
+        ["relation", "isCitedBy"],
+        ["scheme", "doi"],
+      ],
+    };
+  };
+
+  searchApi = () => {
+    return new InvenioSearchApi({
+      axios: {
+        url: ZENODO_BROKER_ENDPOINT,
+        timeout: 5000,
+        headers: {
+          Accept: "application/json",
+        },
+      },
+    });
+  };
```

Comment:
I would suggest to move this config to a separate file for better readability, since it configures axios